libs/@guardian/libs/src/consent-management-platform/sourcepoint.ts

```diff
@@ -134,11 +135,11 @@ export const init = (framework: ConsentFramework, pubData = {}): void => {
 
 					log('cmp', `onMessageChoiceSelect choice_id: ${choice_id}`);
 					log('cmp', `onMessageChoiceSelect choice_type_id: ${choiceTypeID}`);
+					// https://documentation.sourcepoint.com/web-implementation/web-implementation/multi-campaign-web-implementation/event-callbacks#choice-type-id-descriptions
 					if (
-						// https://documentation.sourcepoint.com/web-implementation/web-implementation/multi-campaign-web-implementation/event-callbacks#choice-type-id-descriptions
-						choiceTypeID === 11 ||
-						choiceTypeID === 13 ||
-						choiceTypeID === 15
+						Object.values(SourcePointChoiceTypes).some(
+							(spChoiceType) => spChoiceType === choiceTypeID,
+						)
```

Comment:
I don't really have a ton ts experience, but wouldn't something like this work ?
```typescript
if ( choiceTypeID  in SourcePointChoiceTypes) {
...
}
```

Reply:
![Screenshot 2024-12-17 at 13 40 04](https://github.com/user-attachments/assets/aa88039a-2396-4ef2-80ea-7bc1f6ca1977)

Yeah unfortunatley this doesn't work. As you cant use a type as a value.

Reply:
ah you are right,  I see now that I made a mistake in the way I defined the enum in  my scratchpad test.
I wonder what we get from enums if we are never using it as a type anywhere though.

Unless we are passing around variables of type `SourcePointChoiceTypes` wouldn't it make more sense to have regular number constants ?
Maybe applying some scala thinking to a typescript project but I would only bother with enums if I'm planning to convert the numbers we get from SourcePoint into values of the enum and pass that around.